avm/res/network/network-watcher/tests/e2e/max/main.test.bicep

```diff
@@ -0,0 +1,168 @@
+targetScope = 'subscription'
+
+metadata name = 'Using large parameter set'
+metadata description = 'This instance deploys the module with most of its features enabled.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'NetworkWatcherRG' // Note, this is the default NetworkWatcher resource group. Do not change.
+
+@description('Optional. The location to deploy resources to.')
+param resourcelocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'nnwmax'
+
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+#disable-next-line no-hardcoded-location // Disabled as the default RG & location are created in always one location, but each test has to deploy into a different one
+var testLocation = 'australiaeast'
+var resourceGroupLocation  = 'eastus'
```

Comment:
Why a `resourceGroupLocation`? I would think you can just use the already existing `resourceLocation` parameter for the resource group deployment :)

Reply:
`resourcLocation` can't be used from the randomized locations since our sub has always leftover NetworkWatcherRGs and will always have due to automatic creation required by network resources being created by other e2e's. If NetworkWatcherRG already exists, it lets the workflow fail and doesn't remove it either. One thing I can do is to set it to `westeurope` instead of `eastus` to use our already exsiting networkwatcher but still requires it so set it statically :)

Reply:
Ah, fair point. I wonder if we should solve this rather on the LocationRotation solution's side, than the individual test though. I presume now, because the `resourceLocation` parameter isn't used we'd see a warning upon deployment. However, removing it 'may' also not work (though we can absolutely test this). 
@eriqua & @ChrisSidebotham what are your thoughts?

Reply:
Regarding the existing NetworkWatcher RG - we can clean those up. The auto-creation is a bit of a pain, now that we rotate locations left & right

Reply:
> Yes, NW + location rotation is a tough one 😄 Currently, (and please @ChrisSidebotham correct me otherwise) as part of the location rotation, we only rotate 3 or 4 regions. On top of those, for a few resources, we hardcode a specific location in the test files. Short term I guess for NW we could do the same (hardcode location in tests), but make sure we don't use any location used by other resources to not run into conflicts. Long term, we can discuss about phase 2 of the location rotation feature, to track all regions in a single json file instead of spread across individual test files.

Can we do this now?

Reply:
@eriqua - You are absolutely correct, and we agreed that for resources that need to 'override' the `resourceLocation` this should be done using the test files directly. To make the sure the param is used I was use a var and do a conditional check on it 
`var resourceGroupLocation = resourceLocation == 'uksouth' ? 'uksouth' : 'ukwest'`

Last we discussed this with @jtracey93  we decided against using a single json for 'enforcedLocations', I can't recall why but happy for us to look at this again

Reply:
Agreed, adjusted.

Reply:
Looks good to me @AlexanderSehr are you happy to close this out?

Reply:
> @eriqua - You are absolutely correct, and we agreed that for resources that need to 'override' the `resourceLocation` this should be done using the test files directly. To make the sure the param is used I was use a var and do a conditional check on it `var resourceGroupLocation = resourceLocation == 'uksouth' ? 'uksouth' : 'ukwest'`
> 
> Last we discussed this with @jtracey93 we decided against using a single json for 'enforcedLocations', I can't recall why but happy for us to look at this again

All good @ChrisSidebotham, probably I'm not aligned on that. I thought it was already planned as long term, depending on how many modules would have required an exception. Let's discuss that offline maybe.

To the matter at hand, @segraef aren't we getting conflicts if enforcing the same location for the 3 different tests? Also @ChrisSidebotham @AlexanderSehr which location should we enforce? I thought westeurope was a no go due to capacity issues.

Reply:
West europe should be a no go for compute - networkWatcher in ukwest is fine similar to keyvaults etc. I am happy to keep this in westEurope with @segraef's current changes

Reply:
> West europe should be a no go for compute - networkWatcher in ukwest is fine similar to keyvaults etc. I am happy to keep this in westEurope with @segraef's current changes

True, but we are deploying VMs in both max and waf-aligned test dependencies

Reply:
Remember, the `NetworkWatcherRG` will always remain due to other e2e's running on a daily basis which means it will remain with the first created location. If I run the network-watcher workflow it will remove the NetworkWatcherRG but it will pop up again with the next e2e using a network resource requiring NetworkWatcher for its random location.

Now, here is the problem, due to random locations the next time the network-watcher workflow will fail if the NetworkWatcherRG exists in any other region than defined in the network-watcher e2e (currently westeurope) because only one NetworkWatcherRG can exist per subscription. But also due to leftover resources due to failing workflows we will always have a NetworkWachterRG remaining. It seems we have only one option, to not remove the NetworkWatherRG or create an exemption for the network-watcher.

Reply:
True. I think we all agree that network watcher needs an exemption from the current implementation of resource location rotation. 

The agreement we need is more on which location we should enforce for which test, since we have some limitations. My 4 cents below:
- The location should not be any location with capacity issues. The reason is compute resources, such as virtual machines, are used as deployment test dependencies.
- The location should be different for the different deployment tests. The reason is to avoid conflicts (e.g. one test removes the RG meanwhile the other is still using it to deploy resources).
- The location should not be any location used by other products. That includes the locations we rotate plus any location enforced in other tests as an exception. The reason is same as above, to avoid conflicts.
- Possibly, the implementation should be consistent with others enforcing a particular location.

Reply:
The more I think about it, we might need a daily cleanup job anyways which might partially contribute to our failed pipelines issue currently.

Reply:
Even with my final changes, without a cleanup job, we will always end up with failing workflows on the unique NetworkWatcherRG because, as you mentioned as well already, the removal will try to remove the NetworkWatcherRg fixed to one location while the other test is still deploying into it.

Reply:
That's a different error, which seems not to occur anymore given your last pipeline run :)